File: game/End.py
from .config import *
from .Button import *

class End:
	def __init__(self):
		self.text_font = pg.font.Font(font, int(winHeight * 0.1))
		self.size = [winWidth * 0.2, winHeight * 0.1]
		self.button = Button("BACK TO MENU", (winWidth / 2) - (self.size[0] / 2), winHeight - (self.size[1] * 1.5), self.size[0], self.size[1], winHeight * 0.085)
  
  
	def draw(self, core, win, score):
		if core.custom_mod == "1V1V1V1":
			self.drawSquare(core, win)
		else:
			text = [self.text_font.render(core.players[0].name, True, (255, 255, 255))]
			pos = [[(winWidth / 3) - (text[0].get_size()[0] / 2), (winHeight / 3) - (text[0].get_size()[1] /2)]]
			text.append(self.text_font.render(core.players[0].win, True, (255, 255, 255)))
			pos.append([(winWidth / 3) - (text[1].get_size()[0] / 2), (winHeight / 2) - (text[1].get_size()[1] /2)])
			text.append(self.text_font.render(str(score[0]), True, (255, 255, 255)))
			pos.append([(winWidth / 2) - 75, (winHeight / 2) - (text[2].get_size()[1] /2)])
	
			text.append(self.text_font.render(core.players[1].name, True, (255, 255, 255)))
			pos.append([(winWidth / 3 * 2) - (text[3].get_size()[0] / 2), (winHeight / 3) - (text[3].get_size()[1] /2)])
			text.append(self.text_font.render(core.players[1].win, True, (255, 255, 255)))
			pos.append([(winWidth / 3 * 2) - (text[4].get_size()[0] / 2), (winHeight / 2) - (text[4].get_size()[1] /2)])
			text.append(self.text_font.render(str(score[1]), True, (255, 255, 255)))
			pos.append([(winWidth / 2) + 50, (winHeight / 2) - (text[5].get_size()[1] / 2)])

			text.append(self.text_font.render("_", True, (255, 255, 255)))
			pos.append([(winWidth / 2) - (text[6].get_size()[0] / 2), (winHeight / 2) - (text[6].get_size()[1] * 0.75)])
	
			if core.players.__len__() == 4:
				pos.append(pos[3].copy())
				pos[3][0] = pos[0][0]
				pos[0][0] -= (text[0].get_size()[0] + 50)
				text[4] = self.text_font.render(core.players[2].win, True, (255, 255, 255))
				text.append(self.text_font.render(core.players[2].name, True, (255, 255, 255)))
				text.append(self.text_font.render(core.players[3].name, True, (255, 255, 255)))
				pos.append([pos[7][0] + text[7].get_size()[0] + 50, pos[7][1]])

			for i in range(text.__len__()):
				win.blit(text[i], pos[i])

		self.button.draw(win)

	def drawSquare(self, core, win):
		pass
		text = []
		pos = []

		for i in range(text.__len__()):
				win.blit(text[i], pos[i])
  
	def click(self, core, mousePos):
		if self.button.hitbox.collidepoint(mousePos):
			core.state = "menu"
			core.mode = "none"
			core.pause[0] = False
			core.pause[1].freeze = False

File: srcs/django_code/front/js/game/End.js
import { Button } from "./Button.js";
import { canvas, ctx } from "./canvas.js";
import { is_colliding } from "./Hitbox.js";
import { Vec2 } from "./Vec2.js";

export class End {
	constructor() {
		this.size = [canvas.width * 0.2, canvas.height * 0.1];
		this.button = new Button("BACK TO MENU", (canvas.width / 2) - (this.size[0] / 2), canvas.height * 0.8, this.size[0], this.size[1]);
	}

	draw(core, score) {
		if (core.square) {
			this.drawSquare(core, score);
			this.button = new Button("BACK TO MENU", canvas.width * 0.77, canvas.height * 0.85, this.size[0], this.size[1]);
		}
		else {
			let text = []
			for (let i = 0; i < 2; i++) {
				text.push(core.players[i].name);
				text.push(core.players[i].win);
				text.push(score[i]);
			}
			let pos = [[canvas.width / 3, canvas.height / 3],
					[canvas.width / 3, canvas.height / 2],
					[canvas.width * 0.45, canvas.height / 2],
					[canvas.width / 3 * 2, canvas.height / 3],
					[canvas.width / 3 * 2, canvas.height / 2],
					[canvas.width * 0.55, canvas.height / 2]];
			
			if (core.players.length === 4) {
				pos.push([pos[3][0], pos[3][1]]);
				pos[3][0] = pos[0][0];
				pos[0][0] -= this.size[0];
				text[4] = core.players[2].win;
				text.push(core.players[2].name);
				text.push(core.players[3].name);
				pos.push([pos[6][0] + this.size[0], pos[6][1]]);
			}
			text.push("_");
			pos.push([canvas.width / 2, canvas.height * 0.465]);

			ctx.font = Math.floor(canvas.height * 0.1) + "px pong-teko";
			for (let i = 0; i < text.length; i++)
				ctx.fillText(text[i], pos[i][0], pos[i][1]);
			ctx.font = Math.floor(canvas.height * 0.085) + "px pong-teko";
		}
		this.button.draw();
	}

	drawSquare(core, score) {
		let text = [];
		for (let i = 0; i < core.players.length; i++) {
			text.push(core.players[i].name);
			text.push(core.players[i].win);
			text.push(score[i]);
		}

		let pos = [[canvas.width / 5, canvas.height / 2],
					[canvas.width / 5 * 2, canvas.height / 2],
					[canvas.width * 0.47, canvas.height / 2],
					[canvas.width / 5 * 4, canvas.height / 2],
					[canvas.width / 5 * 3, canvas.height / 2],
					[canvas.width * 0.53, canvas.height / 2],
					[canvas.width / 2, canvas.height / 8],
					[canvas.width / 2, canvas.height / 4],
					[canvas.width / 2, canvas.height * 0.4],
					[canvas.width / 2, canvas.height / 8 * 7],
					[canvas.width / 2, canvas.height / 4 * 3],
					[canvas.width / 2, canvas.height * 0.6]];

		ctx.font = Math.floor(canvas.height * 0.1) + "px pong-teko";
		for (let i = 0; i < text.length; i++)
			ctx.fillText(text[i], pos[i][0], pos[i][1]);
		ctx.font = Math.floor(canvas.height * 0.085) + "px pong-teko";
	}

	click(core, mousePos) {
		const pos = new Vec2(mousePos[0], mousePos[1]);
		if (is_colliding(pos, [0, 0], this.button.hitbox.pos, this.size)) {
			core.state = "menu";
			core.mode = "none";
			core.pause[0] = false;
			core.pause[1].freeze = false;
			core.start_screen = false;
			core.wait_screen = false;
			core.custom_menu = false;
			core.ai = [];
			core.max_score = 10;
			core.online = false;
		}
	}

	responsive() {
		this.size = [canvas.width * 0.2, canvas.height * 0.1];
		this.button = new Button("BACK TO MENU", (canvas.width / 2) - (this.size[0] / 2), canvas.height * 0.8, this.size[0], this.size[1]);
	}
}